tests/main/interfaces-timezone-control/task.yaml

```diff
@@ -0,0 +1,63 @@
+summary: Check that timezone interface works
+
+details: |
+    This test makes sure that a snap using the timezone-control interface
+    can access timezone information and update it.
+
+prepare: |
+    # Install a snap declaring a plug on timezone-control
+    cp $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml.orig
+    sed 's/\[\]/\[timezone-control\]/g' -i $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml
+    snap try $TESTSLIB/snaps/test-snapd-timedate-control-consumer
+
+restore: |
+    # Restore the timedate-control-consumer snap.yaml
+    cp $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml.orig $TESTSLIB/snaps/test-snapd-timedate-control-consumer/meta/snap.yaml
+
+    # Restore the initial timezone
+    if [ -f timezone.txt ]; then
+        timedatectl set-timezone "$(cat timezone.txt)"
+        rm -f timezone.txt
+    fi
+
+execute: |
+    CONNECTED_PATTERN=":timezone-control +test-snapd-timedate-control-consumer"
+    DISCONNECTED_PATTERN="\- +test-snapd-timedate-control-consumer:timezone-control"
+
+    . $TESTSLIB/dirs.sh
+
+    # Check the interface is disconnected by detault
+    snap interfaces | MATCH "$DISCONNECTED_PATTERN"
+
+    # Connect the interfaces for both snaps
+    snap connect test-snapd-timedate-control-consumer:timezone-control
+    snap interfaces | MATCH "$CONNECTED_PATTERN"
+
+    # Read timezones information should be allowed
+    timezone1=$(test-snapd-timedate-control-consumer.timedatectl list-timezones | sed -n 1p)
+    timezone2=$(test-snapd-timedate-control-consumer.timedatectl list-timezones | sed -n 2p)
+
+    # Save the default timezone to be restored at the end
+    echo $(test-snapd-timedate-control-consumer.timedatectl status | grep 'Time zone:' | sed 's/.*Time zone: \(.*\) (.*)/\1/') > timezone.txt
```

Comment:
I think echo (..) is not necessary here? `test-snapd-timedate-control-consumer.timedatectl status | grep ... | sed ... > timezone.txt` should do it.

Reply:
One could probably do with this little grep abomination: 
```
#pcre to the rescue!!
timedatectl status | grep -oP 'Time zone: \K(.*)(?= \()'  
```